### ISSUE TITLE
In-Game CustomClass System

### ISSUE THREAD

**Issue by jhs232** [alkarinv/BattleArena#154]
I was wondering if we could have a system, where players in-game can make their own class . I've already thought the idea through so I think it's decent, but there is probably something that can be improved. This system is meant for people (like me) that want to set up a server, say with guns, and they might have a 100 different guns, and they don't want to have 100 different classes. This idea is where there are 1,2,3 or more "slots" that they can fill. A slot can be filled by a combination of items, set up by the owner. It's quite hard to explain, but let  me just show you what I thought the config could be like. You should be able to specify if the CustomClass is per-team or global.

CustomClass:
  '0': //Slot 0
    '0': //Combination 1
      name: //The name of the combination. Will be used on signs.
      items: //List all the items that combination gives you. Can be more than 1 type of items, enchanted, etc.
      effects: //List of all effects that combination gives you.
    '1': //Combination 2
      name:
      items:
      effects:
  '1': //Slot 1
    '0': //Combination 1
      name:
      items:
      effects:
    '1': //Combination 2
      name:
      items:
      effects:
  armor: //Here you choose the armor. There is only 1 slot, in this case called armor.
    '0':
      armor: //List the items to wear here
      items: //Any extra items you might want to have with the armor
      effects: //List of effects that comes with this.
    '1':
      armor:
      items:
      effects:

Now, this would also be required to be set up in-game. First, you find an area where you want the users to be able to create their own classes. Then you make a wall, and place 1 sign for each slot. You then select them (maybe with a wand of some sort?) and then do /ba cc set ss (ss is short for slot-signs). Then you do /ba cc set global/teamname (this defines if it's global or only for one team), and finally you go to the place you want them to be teleported to when they want to go and customize their class, you do this by doing  /ba cc set tp (for example), and when they do /ba cc tp <teamname> they get teleported there. This means that the set of signs you set up, is associated with that teleport. They could  just do /spawn or whatever other teleport command they have access to, to get out. Now, you go to your lobby at your arena and you place a sign and on the first line you put CustomClass (for example). And then players go back to the place where they customize their own class (CustomClass is per-player, one player may have set it up with Combination 2 in slot 1, and Combination 2 in slot 2, and another might have set it up with Combination 1 in slot 1 and Combination 2 in slot 2) and they scroll through the combinations (they can only choose 1 combination for each slot, you can't have multiple combinations per slot) for each class, by hitting the signs and right clicking. Hitting scrolls forward, right clicking scrolls backwards. If a player selects Combination 1 in Slot 2, it will show the following on the Slot 2 sign:

Slot 2
======                                                This is just a filler
Name of the Combination,                       The name goes down to 4th line if there's not 
defined in the config                                enough space on 3rd line.

Now, when they go in the lobby and choose CustomClass, they will get the items from the combinations they have choosen. Finally, there's permissions. They would probably be something like: 

battlearena.customclass.slot1.combination1
battlearena.customclass.slot1.combination2
battlearena.customclass.slot2.combination1
battlearena.customclass.slot2.combination2
battlearena.customclass.armor.combination1
battlearena.customclass.armor.combination2

Thank you for reading. Please tell me if you do not understand (I'm not the best at explaining).

**Comment by alkarinv** [comment]
So, basically, players can set up their own classes? 

So say like Knight.

Then within a certain range of items for knight... people could pick what they wanted in that slot?

**Comment by jhs232** [comment]
Yes, correct. Instead of the admin having to make a 1000 cross over classes, they let people make their own class within a range of combinations of items. I've also got a few more ideas: People can have any amount of CustomClasses they want, and there should be a amor section for each. Thank you for reading.

**Comment by alkarinv** [comment]
I'll see what I can do, but probably will be next week or so before I can look at it.

**Comment by jhs232** [comment]
Thank you for thinking about implementing this. Thank you so much!
